### PR TITLE
feat: add debug APIs to view and set deck card order

### PR DESCRIPTION
**Pull request by scolley31**
## Summary
- 前端測試時需要控制牌堆順序，新增兩支 debug API
- 不需要 UseCase / Presenter / WebSocket — 純 CRUD，獨立 `DebugController`
- 無 `@Profile` gate，永遠可用，靠 `/api/debug/` 路徑區隔

### 查看牌堆
```
GET /api/debug/games/{gameId}/deck
```
Response:
```json
{
  "gameId": "my-id",
  "deckSize": 78,
  "cardIds": ["BS8008", "BH2028", "BH3029", ...]
}
```
`cardIds[0]` = 下一張會被抽的牌。

### 設定牌堆
```
PUT /api/debug/games/{gameId}/deck
```
Request:
```json
{ "cardIds": ["BS8008", "BH2028", "BH3029"] }
```
- 全量替換整個牌堆
- `cardIds[0]` = 下一張被抽的
- 無效 cardId → 400 `{"message": "Invalid card ID: XYZ"}`

## 實作
- `DebugController.java` — 2 支 endpoint，直接注入 `GameRepository`
- `SetDeckRequest.java` — PUT request DTO
- `DeckResponse.java` — GET/PUT response DTO
- Domain 層零修改

## Test plan
- [x] `mvn install -DskipTests` — 編譯全綠
- [ ] 手動 curl 測試（建遊戲 → GET deck → PUT deck → GET 驗證）

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@RequiredArgsConstructor
+@CrossOrigin
+public class DebugController {
+
+    private final GameRepository gameRepository;
+
+    @GetMapping("/api/debug/games/{gameId}/deck")
+    public ResponseEntity<?> getDeck(@PathVariable String gameId) {
+        Game game = gameRepository.findById(gameId)
+                .orElseThrow(() -> new RuntimeException("Game not found: " + gameId));
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java:26]: **[Nit]** 這裡 throw `RuntimeException` 會被 `LegendsOfTheThreeKingdomsAdvice` 的 `@ExceptionHandler({RuntimeException.class})` catch 並回傳 400。但語意上「Game not found」更適合 404。

可以改用 codebase 已有的 `NotFoundException`：
```java
import com.gaas.threeKingdoms.exception.NotFoundException;
...
.orElseThrow(() -> new NotFoundException("Game not found: " + gameId));
```

`NotFoundException` 已經有對應的 `@ExceptionHandler` 回傳 404（`LegendsOfTheThreeKingdomsAdvice:41-43`）。PUT 也同理。

File: LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            return ResponseEntity.badRequest().body(Map.of("message", "cardIds is required"));
+        }
+
+        // 反序 push，讓 cardIds[0] 在 stack top（下一張抽的）
+        Stack<HandCard> stack = new Stack<>();
+        for (int i = cardIds.size() - 1; i >= 0; i--) {
+            String id = cardIds.get(i);
+            HandCard card = PlayCard.findById(id);
+            if (card == null) {
+                return ResponseEntity.badRequest().body(Map.of("message", "Invalid card ID: " + id));
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java:54]: **[Observation]** 當 invalid card ID 出現在列表中間時，前面的 card 已經被 push 到 stack 但最終不會被使用（因為直接 return 400）。因為 stack 是 local variable，不影響 game state — 沒有副作用，邏輯正確。

只是 call out 這是 "validate during construction" 而非 "validate first then construct" 的 pattern。如果 list 很大且 `findById` 有 side effect（目前沒有），會有問題。目前完全安全。

File: LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            stack.push(card);
+        }
+
+        Deck newDeck = new Deck();
+        newDeck.setCardDeck(stack);
+        game.setDeck(newDeck);
+        gameRepository.save(game);
+
+        return ResponseEntity.ok(new DeckResponse(gameId, cardIds.size(), cardIds));
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/controller/DebugController.java:64]: **[Nit]** 這裡 response 直接用 `cardIds`（request 的 input），而不是從 `stack` 反讀。如果 `findById` 對某些 card ID 做了 normalize（例如大小寫），response 會跟實際 deck 不一致。

目前 `findById` 是 exact match 所以沒問題。但更防禦性的做法是從 stack 反讀（跟 GET 一樣的邏輯），保證 response 反映真實狀態。

File: LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/e2e/DebugDeckApiTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                        .content("""
+                                { "cardIds": ["BH4030", "BS9009"] }
+                                """))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.deckSize").value(2))
+                .andExpect(jsonPath("$.cardIds[0]").value("BH4030"))
+                .andExpect(jsonPath("$.cardIds[1]").value("BS9009"));
+
+        // 驗證 GET 也反映新順序
+        mockMvc.perform(get("/api/debug/games/" + gameId + "/deck"))
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/e2e/DebugDeckApiTest.java:57]: **[Observation / Good]** 👍 PUT 後立刻 GET 驗證 round-trip 是關鍵 test — 確保 `repository.save(game)` 後 MongoDB 的牌堆狀態跟 API 回傳一致。如果 `DeckData` 的序列化有 bug（例如沒保留 stack 順序），這個 test 會 catch 到。

File: LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/e2e/DebugDeckApiTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                .andExpect(status().is4xxClientError());
+    }
+
+    private void givenGameWithKnownDeck() {
+        Player playerA = createPlayer("player-a", 4, General.劉備, HealthStatus.ALIVE, Role.MONARCH);
+        Player playerB = createPlayer("player-b", 4, General.劉備, HealthStatus.ALIVE, Role.TRAITOR);
+        Player playerC = createPlayer("player-c", 4, General.劉備, HealthStatus.ALIVE, Role.REBEL);
+        Player playerD = createPlayer("player-d", 4, General.劉備, HealthStatus.ALIVE, Role.MINISTER);
+        Game game = initGame(gameId, Arrays.asList(playerA, playerB, playerC, playerD), playerA);
+        com.gaas.threeKingdoms.handcard.Deck deck = new com.gaas.threeKingdoms.handcard.Deck();
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/e2e/DebugDeckApiTest.java:120]: **[Nit]** `com.gaas.threeKingdoms.handcard.Deck` 使用 fully-qualified name 是因為跟其他 import 衝突？看起來沒有衝突 — 可以直接 `import com.gaas.threeKingdoms.handcard.Deck;` 再用 `new Deck()`。

**Review by scolley31 — COMMENTED** [review]
# Code Review: Debug Deck API（牌堆查看/設定）

## 整體評價
乾淨的小 feature — 3 個新檔 + 6 個 e2e tests，domain 層零修改。`DebugController` 直接注入 `GameRepository` 跳過 UseCase/Presenter 是正確的決策，因為這是 CRUD debug 工具，不需要 domain events 或 WebSocket push。測試覆蓋完整，包含 happy path、error case、edge case（empty list、duplicate IDs）。

## 缺少的測試場景

無明顯遺漏。目前的 6 個 tests 涵蓋了：GET 正常、PUT+GET round-trip、無效 ID、空陣列、重複 ID、game not found。功能很簡單，這些場景足夠。

## 總結

| 項目 | 評級 |
|------|------|
| 功能正確性 | 🟢 良好 |
| Pattern 一致性 | 🟢 良好（獨立 Controller，不污染 GameController） |
| 程式碼品質 | 🟢 良好（精簡 66 行，無多餘程式碼） |
| 測試品質 | 🟢 良好（6 個 test，覆蓋所有路徑） |

**結論**：✅ 可以 merge，沒有阻塞問題。下方 comments 都是 Nit/Observation 級別。

🤖 Generated with [Claude Code](https://claude.com/claude-code)
